### PR TITLE
fix: fix data is not deleted after alter table

### PR DESCRIPTION
**Pull request by ZuoTiJia**

### CODE REVIEW

File: query_server/query/src/execution/ddl/alter_table.rs [hunk 1]
```diff
@@ -31,28 +32,63 @@ impl DDLDefinitionTask for AlterTableTask {
...
 
-        match &self.stmt.alter_action {
-            AlterTableAction::AddColumn { table_column } => schema.add_column(table_column.clone()),
-            AlterTableAction::DropColumn { column_name } => schema.drop_column(column_name),
+        let req = match &self.stmt.alter_action {
+            AlterTableAction::AddColumn { table_column } => {
+                let mut table_column = table_column.to_owned();
+                table_column.id = schema.next_column_id();
+                schema.add_column(table_column.clone());
+
```
Comment (Subsegment) [inline query_server/query/src/execution/ddl/alter_table.rs:47]: 1

File: tskv/src/kvcore.rs [hunk 2]
```diff
@@ -768,9 +801,9 @@ impl Engine for TsKv {
         new_column: TableColumn,
     ) -> Result<()> {
         let db = self.get_db(tenant, database).await?;
-        db.read()
-            .await
-            .change_table_column(table, column_name, new_column)?;
+        let sid = db.read().await.get_sids(table).await?;
+        self.change_series_schema_column(tenant, database, &sid, column_name, new_column)
+            .await?;
```
Comment (Subsegment) [inline tskv/src/kvcore.rs:806]: 2

File: tskv/src/kvcore.rs [hunk 3]
```diff
@@ -395,6 +395,76 @@ impl TsKv {
...
+        Ok(())
+    }
+
+    async fn add_series_schema_column(
+        &self,
+        tenant: &str,
+        database: &str,
+        series_ids: &[SeriesId],
+        new_column: TableColumn,
+    ) -> Result<()> {
```
Comment (Subsegment) [inline tskv/src/kvcore.rs:424]: 3

File: tskv/src/tseries_family.rs [hunk 4]
```diff
@@ -761,6 +762,22 @@ impl TseriesFamily {
...
 
+    pub fn change_column(&self, sids: &[SeriesId], column_name: &str, new_column: &TableColumn) {
+        self.mut_cache
+            .read()
+            .change_column(sids, column_name, new_column);
+        for memcache in self.immut_cache.iter() {
+            memcache.read().change_column(sids, column_name, new_column);
+        }
+    }
+
```
Comment (Subsegment) [inline tskv/src/tseries_family.rs:773]: 4

**Review by Subsegment — DISMISSED** [review]

**Review by roseboy-liu — APPROVED** [review]

**Pull request merged**